crates/revm/revm-inspectors/src/tracing/builder/geth.rs

```diff
@@ -147,4 +149,75 @@ impl GethTraceBuilder {
             }
         }
     }
+
+    ///  Returns the accounts necessary for transaction execution.
+    ///
+    /// The prestate mode returns the accounts necessary to execute a given transaction.
+    /// diff_mode returns the differences between the transaction's pre and post-state.
+    ///
+    /// * `state` - The state post-transaction execution.
+    /// * `diff_mode` - if prestate is in diff or prestate mode.
+    /// * `db` - The database to fetch state pre-transaction execution.
+    pub fn geth_prestate_traces<DB>(
+        &self,
+        ResultAndState { state, .. }: &ResultAndState,
+        PreStateConfig { diff_mode }: PreStateConfig,
+        db: DB,
+    ) -> Result<PreStateFrame, DB::Error>
+        where DB: DatabaseRef {
+
+        let account_diffs: Vec<_> = state.into_iter().map(|(addr, acc)| (*addr, &acc.info)).collect();
+
+        if diff_mode.unwrap_or_default() {
```

Comment:
this is better,

now I'd like to make this into a function of `PreStateConfig` like

```
fn is_diff_mode(&self) {self.diff_mode.unwrap_or_default()}
``` 

just to improve readability